src/YEthStakerStrategy.sol

```diff
@@ -401,4 +411,51 @@ contract YEthStakerStrategy is BaseStrategy, CustomStrategyTriggerBase {
         require(_token != address(asset), "!asset");
         ERC20(_token).safeTransfer(GOV, ERC20(_token).balanceOf(address(this)));
     }
+
+    /**
+     * @notice Set the trade factory contract address.
+     * @dev For disabling set address(0).
+     * @param _tradeFactory The address of the trade factory contract.
+     */
+    function setTradeFactory(address _tradeFactory) external {
+        require(msg.sender == GOV, "!GOV");
+        _setTradeFactory(_tradeFactory, address(asset));
+    }
+
+    /**
+     * @notice Add a reward token for swapping using TradeFactorySwapper.
+     * @dev Only management can call it.
+     * @param _from The address of the token to swap from. Reward token.
+     * @param _to The address of the token to swap to. Asset token, yETH or st_yETH.
+     */
+    function addRewardTokenForSwapping(
```

Comment:
let's require that _from is not asset, yETH or styETH, so we make sure that trade handler isn't allowed to touch the asset and the other tokens we are using and just can withdraw rewards

Reply:
Fixed: https://github.com/newmickymousse/yeth-strategy/pull/3/files#diff-84b201488eaff9a264559f553c6629871e7e947dbd698d066d403365345d5c33R436

Tested: https://github.com/newmickymousse/yeth-strategy/pull/3/files#diff-ae22ab1dbdffddb7f6f4b116fb5ed1afec50aede6bda7690f3634d0f779718c0R135